app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
Hmm, I was hoping that `finish` would still interrupt the code-path at some point. I guess it can only do so after `onResume` releases control.

I don't have any better ideas. I forget, what was the importance of stopping activity setup code from continuing to run when the session expires?